lib/block-supports/layout.php

```diff
@@ -596,13 +596,16 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 
 	/**
 	 * If columnSpan is set, and the parent grid is responsive, i.e. if it has a minimumColumnWidth set,
-	 * the columnSpan should be removed on small grids.
+	 * the columnSpan should be removed on small grids. If there's a minimumColumnWidth, the grid is responsive.
+	 * But if the minimumColumnWidth value wasn't changed, it won't be set. In that case, if columnCount doesn't
+	 * exist, we can assume that the grid is responsive.
 	 */
-	if ( isset( $block['attrs']['style']['layout']['columnSpan'] ) && isset( $block['attrs']['style']['layout']['parentColumnWidth'] ) ) {
+	if ( isset( $block['attrs']['style']['layout']['columnSpan'] ) && ( isset( $block['parentLayout']['minimumColumnWidth'] ) || ! isset( $block['parentLayout']['columnCount'] ) ) ) {
```

Comment:
Should we also check that `$block['parentLayout']['type']` is set and is `grid`? This is potentially edge casey, but if I take a block that is a child of a grid layout and then drag it to be within another group (i.e. within a Row block elsewhere on the page), then the block still has `columnSpan` set, and the `parentLayout` does not have `columnCount` (because the parent layout is `flex`), but this condition still passes, which results in rules being output that likely shouldn't be:

<img width="883" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/06ae291c-25e6-47b7-96c0-090c3d321af9">

Reply:
Good question! So I opted for not doing that because `$block['parentLayout']['type']` isn't always set. If grid is defined as the default layout type for a block, and no customisations are made, `$block['parentLayout']` will be empty. 
I don't think it's a huge issue because the rules we're applying to children (of both flex and grid) are specific to that layout type, so there shouldn't be any weird side-effects.

Reply:
Sounds good! Yeah, these rules don't wind up affecting anything visually, so seems fine to leave it for now 👍